open-apparel-registry/open-apparel-registry
# Log requests made with token authentication

## Overview

Log all token-authenticated requests to the database for reporting purposes.

Connects #637

## Notes

This PR also includes a migration mistakenly omitted from a previous PR (5a66f74).

We do not want a failure in the request logging to disrupt the handling of a response to we catch all exceptions and send them to Rollbar, failing silently if the call to Rollbar, in turn, raises an exception.

## Testing Instructions

This assumes you have fixtures loaded and processed.

* Run `./scripts/update` and `./scripts/server`
* Run `./scripts/manage dbshell`
* Log in to the web app as `c2@example.com`. 
* Browse http://localhost:6543/profile/2 and generate an API token.
* Browse http://localhost:6543/lists
* Run `select * from api_requestlog;` in `dbshell`. Verify that the session authenticated web app request for the facility lists was _not_ logged.
*  Make an authenticated request with the token and verify that it succeeds.
```
curl \
  -H "Authorization: Token {{API-TOKEN}}" \
  http://localhost:6543/api/facility-lists/
```
* Run `select * from api_requestlog;` in `dbshell`. Verify that the token authenticated web app request for the facility lists _was_ logged.
*  Make an authenticated request for a resource that returns 404.
```
curl \
  -H "Authorization: Token {{API-TOKEN}}" \
  http://localhost:6543/api/facility-lists/3/
```
* Run `select * from api_requestlog;` in `dbshell`. Verify that the 404 request was also logged.

## Checklist
